### PR TITLE
Stepper navigation

### PR DESCRIPTION
**Pull request by YavorLalev**
We want to implement a vertical stepper (navigation bar) which allows the user to progress through each step in sequence.

[US - Stepper-navigation](https://github.com/users/catdieval/projects/1/views/1?pane=issue&itemId=55073903)

### CODE REVIEW

File: Components/ListOfCharts/index.js [hunk 1]
```diff
@@ -10,24 +10,24 @@ export default function ListOfCharts({
...
-                  name={name}
-                  clickedChartType={clickedChartType}
-                  onSelectChartType={() => onSelectChartType(name)}
-                />
-              );
-            })}
-          </Container>
-        </>
-      )}
+      {/* {keynames.length > 0 && ( */}
```
Comment (CJAkkuec) [inline Components/ListOfCharts/index.js:13]:
```suggestion
```

File: Components/ListOfCharts/index.js [hunk 2]
```diff
@@ -10,24 +10,24 @@ export default function ListOfCharts({
...
+      <>
+        {/* <Heading>Step 2: Select the type of chart</Heading> */}
```
Comment (CJAkkuec) [inline Components/ListOfCharts/index.js:15]:
```suggestion
```

File: Components/ListOfCharts/index.js [hunk 3]
```diff
@@ -10,24 +10,24 @@ export default function ListOfCharts({
...
+                icon={icon}
+                name={name}
+                clickedChartType={clickedChartType}
+                onSelectChartType={() => onSelectChartType(name)}
+              />
+            );
+          })}
+        </Container>
+      </>
+      {/* )} */}
```
Comment (CJAkkuec) [inline Components/ListOfCharts/index.js:30]:
```suggestion
```

File: Components/Navigation/styledNavigation.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+export const StepperContainer = styled.section`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin-left: 0.3rem;
+`;
+
+export const SingleStepContainer = styled.section`
+  display: flex;
+  flex-direction: row;
```
Comment (CJAkkuec) [inline Components/Navigation/styledNavigation.js:21]:
```suggestion
```

Comment (CJAkkuec) [reply Components/Navigation/styledNavigation.js:21]: Since `row` is the default value, it doesn't need to be defined.

**Review by CJAkkuec — APPROVED** [review]
Nice job! Since most of my suggestions are about removing commented out code, you can feel free to proceed and don't need to post an updated PR.

As a general sidenote:

If you ever catch yourself writing functions along the lines of `hasCompletedStep1` through `hasCompletedStep6`, you may want to consider refactoring. There's likely other ways to handle progression.

File: Components/ListOfCharts/index.js (hunk 1 above)
Comment (YavorLalev) [reply Components/ListOfCharts/index.js:13]: Done!

**Comment by YavorLalev** [comment]
Thank you for your review and the suggestions above!

File: Components/ListOfCharts/index.js (hunk 2 above)
Comment (YavorLalev) [reply Components/ListOfCharts/index.js:15]: Done!

File: Components/Navigation/styledNavigation.js (hunk 4 above)
Comment (YavorLalev) [reply Components/Navigation/styledNavigation.js]: Done!

File: pages/steps.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import Navigation from "../Components/Navigation";
+import Plotting from "../Components/Plotting";
+
+export default function AllSteps({
```
[the remark this answers is not in the document]
Comment (YavorLalev) [reply pages/steps.js]: Done!

File: pages/index.js [hunk 6]
```diff
@@ -1,98 +1,5 @@
...
-        hasCompletedStep4={hasCompletedStep4}
-        hasCompletedStep5={hasCompletedStep5}
-        hasCompletedStep6={hasCompletedStep6}
-        settings={settings}
-        onSettingsChange={onSettingsChange}
-      />
-    </div>
-  );
+export default function HomePage() {
+  return <StartPage />;
```
[the remark this answers is not in the document]
Comment (YavorLalev) [reply pages/index.js]: We have implemented this one!

File: pages/_app.js [hunk 7]
```diff
@@ -134,41 +123,45 @@ export default function App({ Component, pageProps }) {
...
-  function handleHasCompletedStep5() {
-    alert("Title of the chart is assigned.");
-    setHasCompletedStep5(true);
+  function handleNext() {
+    if (currentStep === 1) {
+      handleConversion();
+    } else if (currentStep === 3) {
+      handleAssignVariables();
+    } else if (currentStep === 6) {
+      handleHasCompletedAllSteps();
```
[the remark this answers is not in the document]
Comment (YavorLalev) [reply pages/_app.js:143]: The condition for steps 1, 3 and 6 is needed to run the functions inside this condition, which are needed to feed data to the app for the following steps. 

For step 2, 4 and 5, there is no special function to run to feed data to the App for the following steps. Therefore we don't need a condition for these steps.

**Pull request merged**